test/configuration_test.dart

```diff
@@ -64,10 +64,10 @@ Future<void> main([List<String>? args]) async {
   });
 
   test('Configuration get/set schema version', () {
-    final config = Configuration([Car.schema]);
+    final config = LocalConfiguration([Car.schema]);
```

Comment:
I haven't touched all the call sites that use the deprecated `Configuration` ctor. I would prefer to have the PR approved first. Otherwise I risk having to revert all the changes.

Reply:
.. arh.. misread. Because I read the `schemaVersion` later. You asked me to remove `schemaVersion` from `InMemoryConfiguration` and `FlexibleSyncConfiguration` hence, it is also gone from the common interface `Configuration`. So either it is `Configuration.local([Car.schema]) as LocalConfiguration` or `LocalConfiguration([Car.schema])`. The later is just shorter.

Reply:
Re-worked to use "named ctors" (static factory methods)